### PR TITLE
feat: Add support for online events

### LINKED ISSUE #1312: Add Support for online events
### Feature Request

**Feature Description**
Eventbrite has support for adding online events and it can be implemented in Orga App as well.

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by sridharjajoo**
Fixes #1312 

Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

ezgif com-resize 40

### CODE REVIEW

**Comment by sridharjajoo** [comment]
@iamareebjamal @Masquerade0097 @mishuvs Please review

File: app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardPresenter.java [hunk 1]
```diff
@@ -112,7 +112,7 @@ public void confirmToggle() {
         if (Event.STATE_PUBLISHED.equals(event.state)) {
             getView().switchEventState();
             getView().showEventUnpublishDialog();
-        } else if (Utils.isEmpty(event.getLocationName())) {
+        } else if (Utils.isEmpty(event.getLocationName()) && !event.isEventOnline) {
```
Comment (mishuvs) [inline app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardPresenter.java]: @sridharjajoo I think the it should be placed first. If there's no location name, the condition `!event.isEventOnline` will not be checked, but an online event can have a location in the case when an event is made online later.

Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardPresenter.java]: Use function

**Comment by sridharjajoo** [comment]
@iamareebjamal done the required changes

File: app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardPresenter.java [hunk 2]
```diff
@@ -195,6 +195,10 @@ private void loadCheckInTimesChart() {
             return attendeeRepository.getAttendees(getId(), forceReload);
     }
 
+    private boolean isEventOnline() {
```
[the remark this answers is not in the document]
Comment (sridharjajoo) [reply app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardPresenter.java]: Actually I tried using the functions in event such as IsEventOnline( ), getEventOnline( ) but all of them are giving errors regarding method not found

[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardPresenter.java]: Are you using Lombok plugin?

[the remark this answers is not in the document]
Comment (sridharjajoo) [reply app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardPresenter.java]: Yes I am

[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardPresenter.java]: OK, let it be and check why it is happening in other issue

**Pull request merged**